pkg/liqo-controller-manager/resource-request-controller/utils.go

```diff
@@ -68,18 +68,31 @@ func (r *ResourceRequestReconciler) ensureControllerReference(foreignCluster *di
 	return true, controllerutil.SetControllerReference(foreignCluster, resourceRequest, r.Scheme)
 }
 
+// createForeignCluster creates the foreign cluster for the given RR. It tries using the cluster name first, and retries
+// with the UniqueName if there is already a foreign cluster with the same name.
 func (r *ResourceRequestReconciler) createForeignCluster(ctx context.Context,
-	resourceRequest *discoveryv1alpha1.ResourceRequest) (*discoveryv1alpha1.ForeignCluster, error) {
+	resourceRequest *discoveryv1alpha1.ResourceRequest, isRetry bool) (*discoveryv1alpha1.ForeignCluster, error) {
```

Comment:
You should avoid using hard-to-read bool values

You can wrap this function like that
```go
func createForeignCluster(context.Context, *discoveryv1alpha1.ResourceRequest) (*discoveryv1alpha1.ForeignCluster, error) {
  return createForeignClusterWithRetry(ctx, rr, false)
}

func createForeignClusterWithRetry(context.Context, *discoveryv1alpha1.ResourceRequest, isRetry bool) (*discoveryv1alpha1.ForeignCluster, error)
```